src/Analyser/MutatingScope.php

```diff
@@ -1750,7 +1751,11 @@ static function (Node $node, Scope $scope) use ($arrowScope, &$arrowFunctionImpu
 							&& !is_numeric($varValue)
 							&& function_exists('str_increment')
 						) {
-							$varValue = str_increment($varValue);
+							try {
```

Comment:
This feels kinda weird because
- On PHP 8.3 & 8.4 we return ErrorType while it's still valid code (even if deprecated)
- On PHP 8.5+ we're still returning `'1'` for `''++` while it's invalid code

Should we introduce a method inside `PHPVersion` class ?

Then you could keep `@++$varValue` for PHPVersion <= 8.5